From 8747f9fc35cbfe91657149a6091e3629e18afe6c Mon Sep 17 00:00:00 2001
From: Andrey Dolnikov <andrey.dolnikov@cogentembedded.com>
Date: Wed, 7 Oct 2020 18:13:39 +0300
Subject: [PATCH 09/12] gpu: drm: rcar-du: rcar_du_vsp: Check if gem buffer has
 a physical address

Do not copy the sg in case gem buffer does have a physical address.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_vsp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
index 3c34437..9246351 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vsp.c
@@ -203,7 +203,7 @@ int rcar_du_vsp_map_fb(struct rcar_du_vsp *vsp, struct drm_framebuffer *fb,
 		struct drm_gem_cma_object *gem = drm_fb_cma_get_gem_obj(fb, i);
 		struct sg_table *sgt = &sg_tables[i];
 
-		if (gem->sgt) {
+		if (gem->sgt && !gem->paddr) {
 			struct scatterlist *src;
 			struct scatterlist *dst;
 
-- 
2.7.4

